flow/FileTraceLogWriter.cpp

```diff
@@ -87,7 +87,18 @@ void FileTraceLogWriter::write(const std::string& str) {
 void FileTraceLogWriter::open() {
 	cleanupTraceFiles();
 
-	auto finalname = format("%s.%d.%s", basename.c_str(), ++index, extension.c_str());
+	++index;
+	int indexWidth = -1;
+	while (index > 0) {
+		index /= 10;
+		++indexWidth;
+	}
+
+	// this allows one process to write 10 billion log files
+	// this should be enough - if not we could make the base larger...
+	ASSERT(index >= 0 && index < 10);
+
+	auto finalname = format("%s.%d.%d.%s", basename.c_str(), indexWidth, ++index, extension.c_str());
```

Comment:
We can just pass `index` without modification since we are incrementing it above (once we fix the reset to 0 issue):

```suggestion
	auto finalname = format("%s.%d.%d.%s", basename.c_str(), indexWidth, index, extension.c_str());
```

Reply:
Done locally - sorry I rebased locally before seeing this comment - which is why I didn't commit your suggestion but changed it myself.